tests/SerdeTest.php

```diff
@@ -607,6 +609,34 @@ protected function nested_objects_with_flattening_validate(mixed $serialized): v
 
     }
 
+    /**
+    * @test
+    * @group typemap
+    */
+    public function nested_objects_with_flattening2(): void
+    {
+        foreach ($this->formatters as $formatter) {
+            if (($formatter->format() === $this->format) && !$formatter instanceof SupportsCollecting) {
+                $this->markTestSkipped('Skipping flattening tests on non-flattening formatters');
+            }
+        }
+
+        $s = new SerdeCommon(formatters: $this->formatters);
+
+        $data = new NestedFlattenObject2('First', null);
+
+        $serialized = $s->serialize($data, $this->format);
+
+        $this->nested_objects_with_flattening2_validate($serialized);
+
+        // no deserialization as flattening causes the object always to be instantiated even when there is no property
```

Comment:
Eh?  I don't follow.  We need to test deserialization, even if it isn't perfectly symmetric.

Reply:
Ok, test added.